Feature/pt 1214 crowdin integration

ydrisR opened the pull request:
* Add the the javascript bits required to use the Crowdin live preview features in hydromail.
* Also update the locale files (Crowdin export automatically remove empty lines)

ydrisR committed:
Integrate crowdin.com "live preview" feature

ydrisR committed:
Use a dedicated string file for locales

Crowdin does not like using the same file as a source and destination.

ydrisR committed:
Update Crowdin configuration file

ydrisR committed:
translation files update

ydrisR committed:
Latest configuration and notes changes prior to the release

ydrisR committed:
Bump version to 0.8.1

BigFab commented on crowdin.yml:
```diff
@@ -0,0 +1,3 @@
+files:
+  - source: /templates/locales/strings.yaml
```
can we use _/templates/locales/en.yaml_ for the source? so we don't have to maintain 2 files for same purpose?

ydrisR replied:
no, I changed this on purpose.
By the way, we don't need to maintain the file en.yaml which will be maintained from crowdin

BigFab reviewed (commented):
_refreshLocales.go_ shall we change line 5 to be:
```go
// Used to interact with a remote localization service such as Crowdin
```
instead of
```go
// Used to interact with a remote localization service such as LoCo or Lokalize
```

BigFab commented on templates/preview/mailPreview.go:
```diff
@@ -44,7 +44,7 @@ func main() {
 	}
 	router := mux.NewRouter()
-	// localizationManager = NewLocoManager(config.LocalizeServiceUrl, config.LocalizeServiceAuthKey)
+	// localizationManager := NewLokaliseManager(config.LocalizeServiceUrl, config.LocalizeServiceAuthKey)
```
```suggestion
```

BigFab commented on templates/preview/local-env.sh:
```diff
@@ -0,0 +1,11 @@
+export TIDEPOOL_HYDROPHONE_SERVICE="{
```
do we need 2 environment files? env.sh and local-env.sh

ydrisR replied:
we need the local-env.sh for local dev
The other environment file contain what is needed when it is deployed. It is not mandatory as we use ansible to generate a configuration file anyway...
Is it a problem to ship the package with a default env file?

BigFab replied:
I thought we usually have 1 env file that we override based on context (local dev, deployment here or there..)

BigFab commented on templates/preview/env.sh:
```diff
@@ -2,10 +2,10 @@ export TIDEPOOL_HYDROPHONE_SERVICE="{
-    \"localizeServiceUrl\": \"https://localise.biz/api\",
+    \"supportUrl\": \"mailto:yourloops@diabeloop.com\",
+    \"i18nTemplatesPath\": \"./\",
+    \"localizeServiceUrl\": \"xxx\",
```
```suggestion
```

BigFab commented on templates/preview/env.sh:
```diff
@@ -2,10 +2,10 @@ export TIDEPOOL_HYDROPHONE_SERVICE="{
+    \"supportUrl\": \"mailto:yourloops@diabeloop.com\",
+    \"i18nTemplatesPath\": \"./\",
+    \"localizeServiceUrl\": \"xxx\",
     \"localizeServiceAuthKey\": \"xxxxxxx\"
```
```suggestion
```

BigFab commented on templates/preview/env.sh:
```diff
@@ -2,10 +2,10 @@ export TIDEPOOL_HYDROPHONE_SERVICE="{
-    \"i18nTemplatesPath\": \"../\",
-    \"localizeServiceUrl\": \"https://localise.biz/api\",
+    \"supportUrl\": \"mailto:yourloops@diabeloop.com\",
+    \"i18nTemplatesPath\": \"./\",
```
```suggestion
    \"i18nTemplatesPath\": \"./\"
```

BigFab commented on templates/preview/local-env.sh:
```diff
@@ -0,0 +1,11 @@
+    \"webUrl\": \"https://preview.your-loops.dev\",
+    \"patientPasswordResetUrl\": \"https://support.diabeloop.com/hc/articles/360025168234\",
+    \"supportUrl\": \"mailto:yourloops@diabeloop.com\",
+    \"i18nTemplatesPath\": \"../\",
```
```suggestion
    \"i18nTemplatesPath\": \"../\"
```

BigFab commented on templates/preview/local-env.sh:
```diff
@@ -0,0 +1,11 @@
+    \"patientPasswordResetUrl\": \"https://support.diabeloop.com/hc/articles/360025168234\",
+    \"supportUrl\": \"mailto:yourloops@diabeloop.com\",
+    \"i18nTemplatesPath\": \"../\",
+    \"localizeServiceUrl\": \"xxx\",
```
```suggestion
```

BigFab commented on templates/preview/local-env.sh:
```diff
@@ -0,0 +1,11 @@
+    \"supportUrl\": \"mailto:yourloops@diabeloop.com\",
+    \"i18nTemplatesPath\": \"../\",
+    \"localizeServiceUrl\": \"xxx\",
+    \"localizeServiceAuthKey\": \"xxxxxxx\"
```
```suggestion
```

ydrisR committed:
apply PR review suggestions

BigFab commented on crowdin.yml:
```diff
@@ -0,0 +1,3 @@
+files:
```
I understand this file is to be used for Crowdin CLI. Why do we need it here?

BigFab commented:
I finished my review. I still have few questions for you @ydrisR

ydrisR committed:
Update environment files based on Fabrice feedbacks

Keep the local evn var file only.
For test/preview/prod environments the configuration is managed separatly

BigFab reviewed (approved):
LGTM Very good job @ydrisR :+1:
